File: Association Rule Analysis.R
run_association_rules <- function(data, item_column, min_support = 0.001, min_confidence = 0.05, target = "rules") {
  txn <- as(split(data[[item_column]], data$PASSENGER_ID), "transactions")
  rules <- apriori(txn, parameter = list(supp = min_support, conf = min_confidence, target = target))
  return(as(rules, "data.frame"))
}

File: Association Rule Analysis.py
from mlxtend.frequent_patterns import apriori, association_rules

def run_association_rules(data, min_support=0.001, min_confidence=0.05):
    frequent_itemsets = apriori(data, min_support=min_support, use_colnames=True)
    rules = association_rules(frequent_itemsets, metric="confidence", min_threshold=min_confidence)
    return rules
